libs/blog/ad-banner/ui/src/lib/banner-carousel/al-banner-carousel.component.ts

```diff
@@ -0,0 +1,27 @@
+import { ChangeDetectionStrategy, Component, input } from '@angular/core';
+
+import { AdImageBanner } from '../ad-image-banner/ad-image-banner-data.interface';
+import { AdImageBannerComponent } from '../ad-image-banner/ad-image-banner.component';
+import { AlInfiniteSliderDirective } from '../infinite-slider-directive/al-infinite-slider.directive';
+
+@Component({
+  selector: 'al-banner-carousel',
+  standalone: true,
+  imports: [AdImageBannerComponent, AlInfiniteSliderDirective],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="overflow-hidden">
+      <div class="flex">
+        <al-ad-image-banner
+          *alInfiniteSlider="let banner of banners(); msPerSlide: msPerSlide()"
+          class="flex-shrink-0 flex-grow-0 basis-full"
+          [banner]="banner"
+        ></al-ad-image-banner>
+      </div>
+    </div>
+  `,
+})
+export class AlBannerCarouselComponent {
+  readonly banners = input.required<AdImageBanner[]>();
```

Comment:
If there are no probles, we can use protected access modifier

Reply:
that's an input so we need to keep it public